以下には、クイズが何問か並んでいます。よく読んで、選択肢を1つ定めて、メモをとってください。すべてのクイズの解答が決まったら、このページの最後にあるリンクから正解ページを見て、答え合わせをしてください。このクイズの結果は、成績には関係ありません。

しかし、各回の最後に設置されている小テストは、受験1回目の成績が、全体の成績に反映されます。ですから、小テストの前に、このクイズを使って、しっかり理解しておくとよいです。

**QUIZ3-1**

次の選択肢のうち、正しいものを1つ選べ。

①理化学研究所の鵜澤さんは、プログラムを利用することで「複雑な作業が簡単にできる」という利点があると述べていた。

②理化学研究所の鵜澤さんは、PythonのプログラムとC言語を利用したプログラムの説明をしていた。

③理化学研究所の鵜澤さんが利用していた装置では、小型のコンピュータが Python のプログラムに従って動作して、装置を制御していた。

④理化学研究所の鵜澤さんは、プログラムを利用する人間の問題点について述べていた。

**QUIZ3-2**

次のプログラムを動作させた。

In [ ]:
x = int(input('x?'))

if x > 50:
    print('A')
elif x > 25:
    print('B')
else:
    print('C')

このとき、B が出力される入力 x の値として正しいもの選べ。（実行せずに解答せよ。）

① 15

② 45

③ 65

④ 75

**QUIZ3-3**

x = -1 で y = 1 で z = 0 とする。次の選択肢のうち、真となる（成立する）Pythonの条件式を1つ選べ。

① x == 0 or (y == 1 and z == 0)

② x < 3 and y < 1 and z > 0

③ x == 0 and y < 0

④ x < 4, y < 5, z < 5

**QUIZ3-4**

次のプログラムを動作させた。

In [1]:
nishi = 2

def kazoeru(x):
    higashi = x
    return(nishi + higashi)

print(kazoeru(3))

5


このとき、次の選択肢のうち、正しいものを1つ選べ。

① 変数 higashi は大域変数である。

② プログラム終了時に、nishi の値は 5 である。

③ プログラム終了後に、print(higashi) を実行するとエラーになる。

④ 出力される結果は 10 である。

**QUIZ3-5**

次のプログラムを動作させた。

In [2]:
i = 1
n = 10
while i < n:
    print(i, end="")
    i += 4

159

このとき、次の選択肢のうち、出力結果として正しいものを1つ選べ。（実行せずに解答せよ。）

① 159

② 15913

③ 04812

④ 012345678910

第3回のクイズの解答（正解）はこちらです。 https://colab.research.google.com/drive/136tP28FBy4UbJVlxU2hiZnV2x3e4jYBd